OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -56,19 +56,25 @@ static string[] QueryDevices(string label, int type)
 
 			var devices = new List<string>();
 			var next = ALC10.alcGetString(IntPtr.Zero, type);
+			if (next == IntPtr.Zero || AL10.alGetError() != AL10.AL_NO_ERROR)
+			{
+				Log.Write("sound", "Failed to query OpenAL device list using {0}", label);
+				return new string[] { };
+			}
+
 			do
 			{
 				var str = Marshal.PtrToStringAnsi(next);
+				if (string.IsNullOrEmpty(str))
```

Comment:
referencesource.microsoft.com/#mscorlib/system/runtime/interopservices/marshal.cs,124 confirms the behaviour of `PtrToStringAnsi`.  Recent versions of mono use the same source.